ui/src/components/build/CreateBuildConfigDialog.js

```diff
@@ -0,0 +1,267 @@
+    this.sshAuthState = new SubState(this, 'config.source.sshAuth');
+    this.sourceState = new SubState(this, 'config.source');
+  }
+
+
+  onSaveBuildConfig = () => {
+    const { config } = this.state;
+    const { onSave, update = false } = this.props;
+    onSave && onSave({ update, config });
+  };
+
+  onEnvVarsChange = (rows) => {
+    this.configState.set(
+      'envVars',
+      rows.map(({ key, value }) => ({
+        name: key,
+        value,
+      })),
+    );
+  };
+
+  cancel = () => {
+    const { onCancel } = this.props;
+    onCancel && onCancel();
+  };
+
+  renderFormButtons = (onCancel, update) => (
+    <div>
+      <Button onClick={this.onSaveBuildConfig}>{update ? 'Save' : 'Create'}</Button>
+      <Button onClick={onCancel}>Cancel</Button>
+    </div>
+  );
+
+  renderBuildSection = () => {
+    const { platform = BUILD_PLATFORM_ANDROID, buildType = BUILD_AUTH_TYPE_PUBLIC } = this.buildState.getOrEmpty();
+    return (
+      <div className="section">
+        <h3 className="with-divider">Build Configuration</h3>
+        <FormGroup>
+          <ControlLabel>Platform</ControlLabel>
+          <FormDropdown
+            id="platform"
+            items={[BUILD_PLATFORM_ANDROID, BUILD_PLATFORM_IOS]}
+            titles={['Android', 'iOS']}
+            onSelect={active => this.buildState.set('platform', active)}
+            selected={platform}
+          />
+        </FormGroup>
+        <FormGroup>
+          <ControlLabel>Build Type</ControlLabel>
+          <FormDropdown
+            id="buildtype"
+            items={[BUILD_TYPE_DEBUG, BUILD_TYPE_RELEASE]}
+            titles={['Debug', 'Release']}
+            onSelect={active => this.buildState.set('buildType', active)}
+            selected={buildType}
+          />
+        </FormGroup>
+        <FormGroup>
+          <ControlLabel>Environment Variables</ControlLabel>
+          <KeyValueEditor onChange={this.onEnvVarsChange} />
+        </FormGroup>
+      </div>
+    );
+  };
+
+  renderSourceSection = () => {
+    const { advancedOptions } = this.state;
+    const {
+      gitUrl = '', gitRef = 'master', jenkinsFilePath = '/', authType = BUILD_AUTH_TYPE_PUBLIC,
+    } = this.sourceState.getOrEmpty();
+    return (
+      <div className="section">
+        <h3 className="with-divider">Source Configuration</h3>
+        <Grid fluid className="sourceSection">
+          <Row>
+            <Col lg={advancedOptions ? 8 : 12}>
+              <FormGroup>
+                <ControlLabel className="required">Git Repository URL</ControlLabel>
+                <FormControl type="text" onChange={e => this.sourceState.set('gitUrl', e.target.value)} value={gitUrl} />
+                <HelpBlock>Git URL of the source code to build.</HelpBlock>
+                <HelpBlock>
+            View the
+                  {' '}
+                  <a href="#" onClick={() => this.setState({ advancedOptions: true })}>advanced options</a>
```

Comment:
In OpenShift when you click the `advanced options`, it hides. Maybe we should do the same here.

Reply:
ok